packages/services/server/src/use-unexpected-serialization-error.ts

```diff
@@ -0,0 +1,44 @@
+import { GraphQLScalarType } from 'graphql';
+import { type Plugin } from 'graphql-yoga';
+import { isGraphQLError } from '@envelop/core';
+import { MapperKind, mapSchema } from '@graphql-tools/utils';
+
+/**
+ * Scalar serialization errors are swallowed by graphql-yoga, because they are GraphQLErrors and not Errors.
+ * We can map them to normal errors in order to report them.
+ *
+ * See https://github.com/n1ru4l/envelop/issues/1808
+ */
+export function useUnexpectedSerializationError(): Plugin<any> {
+  return {
+    onSchemaChange({ schema, replaceSchema }) {
+      if (schema.extensions?.[didApplyTransformSymbol]) {
+        return;
+      }
+
+      const newSchema = mapSchema(schema, {
+        [MapperKind.SCALAR_TYPE](type) {
+          const config = type.toConfig();
+          return new GraphQLScalarType({
+            ...config,
+            serialize(value) {
+              try {
+                return config.serialize(value);
+              } catch (err) {
+                if (isGraphQLError(err)) {
+                  throw new Error(err.message);
+                }
```

Comment:
This makes all scalars plain Errors, are they then reported to the user? Like, what if its the users fault having provided a malformed value?

Reply:
The `serialize` function is only used for resolver values. These do not really matter to the user and could even be a security risk. The malformed values are handled by the `parse` function,